### ISSUE TITLE
Returning model output via `attribute` function

### ISSUE THREAD

**Issue by aarzchan** [pytorch/captum#761]
Hi! I'm interested in regularizing the model via some attribution-based loss. 

Specifically, let L_task be the task loss, L_attr be the attribution loss, and L = L_task + \lambda L_attr be the total loss, where \lambda is a loss weight hyperparameter. Here, L_task is computed from the model's output w.r.t. some target, and L_attr is computed from the attributions w.r.t. some target. 

My goal is to backpropagate from L to update the model's parameters. To achieve this most easily, I think the `attribute` function would need to be able to return the model's output, so that L_task can be computed. According to the API reference, this functionality doesn't seem to be supported, so I was wondering if you could recommend the best workaround.

Thanks for your help!

**Comment by aarzchan** [comment]
After digging a bit, I realized that returning the model output via `IntegratedGradient`'s `attribute` function isn't what I want, since the returned outputs would be the outputs obtained from the interpolated inputs (not from the original input). It would probably be easier to handle L_task outside of the `attribute` function by calling the model forward function directly.

Nonetheless, maybe for other situations, it could be useful to return the outputs from the `attribute` function. Not sure if this is a feature you'd be interested in adding?

**Comment by vivekmig** [comment]
Hi @aarzchan , one hack to accomplish what you're looking for is something like this with a global variable:
```
def wrapper_forward(inp):
  global model_out
  out = model(inp)
  model_out = out
  return out
ig = IntegratedGradients(wrapper_forward)
```

This can also be alternatively implemented with an additional forward arg data structure passed to a wrapper function:
```
def wrapper_forward(inp, output_history):
  out = model(inp)
  output_history.append(out)
  return out
ig = IntegratedGradients(wrapper_forward)
output_history = []
attr = ig.attribute(torch.randn(1,3), target = 1, additional_forward_args=output_history)
```

We likely don't have enough use-cases of this yet to justify adding this to the standard API, but hopefully one of these work-arounds would be sufficient.

In your use-case, you're right, the model output in IntegratedGradients would be the interpolated inputs, so may not include what you need. But if you use 'reimann_right' as your approximation method, the interpolated inputs would include the original input as the last example / batch.

**Comment by aarzchan** [comment]
Hi @vivekmig, thanks for your help!
